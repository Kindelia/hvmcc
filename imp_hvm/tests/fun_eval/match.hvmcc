function test_match (a) {
  {test_match a} = match a {
    0 => Unit
    1 => 1
    a => @x x
  }
}

function empty_rule () {
  {empty_rule} = Unit
}

function Main () {
  {Main} = {Test (empty_rule) (test_match 1) (test_match 2)}
}
