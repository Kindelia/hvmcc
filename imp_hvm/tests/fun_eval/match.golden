{Test Unit 1 Î»x x}